### PR TITLE
Copy paste support

### PR DESCRIPTION
**Pull request by mx4492**

### CODE REVIEW

File: Wily-ios/Wily/WilyPlaylist+Pasteboard.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#import "WilyPlaylist.h"
+
+@interface WilyPlaylist (Pasteboard)
+
+- (void)copyYouTubeLinkToCurrentlyPlayingSong;
```
Comment (ayushgoel) [inline Wily-ios/Wily/WilyPlaylist+Pasteboard.h]: copyYouTubeLink ~~To~~ Of CurrentlyPlayingSong ?

File: Wily-ios/Wily/WilyPlaylist+Pasteboard.m [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  NSDictionary *song = [self songs][idx];
+  if (song == nil) {
+    return;
+  }
+
+  NSString *URLString = WilyYouTubeURLStringFromSong(song);
+  NSLog(@"Setting pasteboard string = %@", URLString);
+  [UIPasteboard generalPasteboard].string = URLString;
```
Comment (ayushgoel) [inline Wily-ios/Wily/WilyPlaylist+Pasteboard.m]: Why not set `URL` instead?
https://developer.apple.com/library/prerelease/ios/documentation/UIKit/Reference/UIPasteboard_Class/index.html#//apple_ref/occ/instp/UIPasteboard/URL

File: Wily-ios/Wily/WilyPlaylist+Pasteboard.m [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+- (void)playSongFromCopiedYouTubeLink {
+  NSString *URLString = [UIPasteboard generalPasteboard].string;
```
Comment (ayushgoel) [inline Wily-ios/Wily/WilyPlaylist+Pasteboard.m]: In context of last comment, can we check the `URL` also in pasteboard?

File: Wily-ios/Wily/WilyPlaylist+Pasteboard.m (hunk 2 above)
Comment (mx4492) [inline Wily-ios/Wily/WilyPlaylist+Pasteboard.m]: Nice!

**Comment by mx4492** [comment]
Updated

File: Wily-ios/Wily/WilyYouTubeURL.h [hunk 4]
```diff
@@ -0,0 +1,6 @@
+@import Foundation;
+
+/// @return nil if @p string is not a valid Youtube Video URL.
+NSDictionary *WilyYouTubeSongFromURLString(NSString *string);
+
+NSString *WilyYouTubeURLStringFromSong(NSDictionary *song);
```
Comment (ayushgoel) [inline Wily-ios/Wily/WilyYouTubeURL.h]: This method crashes if `@"id"` is not in song dictionary. Would adding that in documentation be helpful?

Comment (mx4492) [inline Wily-ios/Wily/WilyYouTubeURL.h]: A "song" dictionary is assumed to contain an "id" field at the very least. In any case, I've hardened the method to remove the assumption.

**Comment by ayushgoel** [comment]
GTM :+1:

**Pull request merged**